### PR TITLE
[in tests] Refactoring Tiled Map System to Support Large Maps

### PR DESCRIPTION
**Pull request by RafaelBarbosatec**
Basically I try not to create the Tile component for the entire map. I only create the Tile component for the elements visible on the screen.

### CODE REVIEW

**Comment by RafaelBarbosatec** [comment]
Hi @KyleKun !
If you can help test and evolve this. it would be very nice!
😊

File: lib/map/map_world.dart [hunk 1]
```diff
@@ -161,4 +157,104 @@ class MapWorld extends MapGame {
       canvas.drawPath(path, _paintPath);
     }
   }
+
+  void _updateTilesToRender() async {
```
Comment (RafaelBarbosatec) [inline lib/map/map_world.dart:161]: Actually this is a biggest performance bottleneck. is very visible running on the web

**Comment by KyleKun** [comment]
Hi @RafaelBarbosatec , sure!
Will do it asap, I've a PR to improve TalkDialog incoming as well 😁

File: lib/map/map_world.dart (hunk 1 above)
Comment (RafaelBarbosatec) [reply lib/map/map_world.dart:161]: It seems to be something of a web debug way. The crash is caused by `Future.forEach`. but running in release mode everything OK

Comment (KyleKun) [reply lib/map/map_world.dart:161]: @RafaelBarbosatec I am testing both on Android and Web and it seems very nice, the performance issue on web is solved running in release mode so I don't think it is a big problem.

The only thing I noticed is that the player movement is lagging if I increase his speed (I am using 300.0), take a look:
https://imgur.com/a/GVtKHwI

Comment (RafaelBarbosatec) [reply lib/map/map_world.dart:161]: Perfect. Thanks so much!
 I'll take a look at that case.

Comment (RafaelBarbosatec) [reply lib/map/map_world.dart:161]: Hi @KyleKun !
I found the problem. I had to make improvements to the collision system!
Try again and tell me what you think.

**Review by KyleKun — APPROVED** [review]
LGTM!
Tested both on Web and Android, it seems to be working as expected.

I caught 2 problems, but I think we can raise separate issues for them since they are not major in my opinion:

1) Light is still laggy like the player was before:
https://imgur.com/a/6KeLsIW

2) Sometimes the collision seems not to work for, check the fireball behavior after I move the player a little:
https://imgur.com/a/nI084r4

But I think we can merge this and work on these problems later in separate PRs.
Let me know what you think. @RafaelBarbosatec

**Pull request merged**